products/BellHybrid/apps/common/include/common/layouts/HomeScreenLayouts.hpp

```diff
@@ -3,22 +3,24 @@
 
 #pragma once
 
-#include "layouts/HomeScreenLayoutClassicWithAmPm.hpp"
-#include "layouts/HomeScreenLayoutClassicWithBattery.hpp"
-#include "layouts/HomeScreenLayoutClassicWithTemp.hpp"
+#include "HomeScreenLayoutClassicWithAmPm.hpp"
+#include "HomeScreenLayoutClassicWithBattery.hpp"
+#include "HomeScreenLayoutClassicWithTemp.hpp"
 
 #include <map>
 #include <string>
 #include <typeindex>
 
 namespace gui
 {
-    using LayoutGenerator = std::function<std::unique_ptr<gui::BaseHomeScreenLayoutProvider>()>;
+    class BaseHomeScreenLayoutProvider;
 
-    std::map<std::string, LayoutGenerator> homeScreenLayouts = {
-        {"ClassicWithTemp", []() { return std::make_unique<gui::HomeScreenLayoutClassicWithTemp>("ClassicWithTemp"); }},
-        {"ClassicWithAmPm", []() { return std::make_unique<gui::HomeScreenLayoutClassicWithAmPm>("ClassicWithAmPm"); }},
-        {"ClassicWithBattery",
-         []() { return std::make_unique<gui::HomeScreenLayoutClassicWithBattery>("ClassicWithBattery"); }},
+    using LayoutGenerator = std::function<BaseHomeScreenLayoutProvider *()>;
+
+    inline std::map<std::string, LayoutGenerator> homeScreenLayouts()
```

Comment:
1. please just create cpp files, this is what these are for (to have declarations)
2. you request explicitly compiler to copy this chunk of code everywhere used,it's not very embedded friendly :)

Reply:
fixed, thanks!